OneSignalSDK/onesignal/src/main/java/com/onesignal/ADMMessageHandlerJob.kt

```diff
@@ -0,0 +1,46 @@
+package com.onesignal
+
+
+import android.content.Context
+import android.content.Intent
+import com.amazon.device.messaging.ADMMessageHandlerJobBase
+
+class ADMMessageHandlerJob : ADMMessageHandlerJobBase() {
+
+    override fun onMessage(context: Context?, intent: Intent?) {
+        val bundle = intent!!.extras
+
+        val processedResult = NotificationBundleProcessor.processBundleFromReceiver(context, bundle)
+        // TODO: Figure out the correct replacement or usage of completeWakefulIntent method
+//      FCMBroadcastReceiver.completeWakefulIntent(intent);
+
+        // TODO: Figure out the correct replacement or usage of completeWakefulIntent method
+//      FCMBroadcastReceiver.completeWakefulIntent(intent);
+        if (processedResult.processed()) return
+
+        val payload = NotificationBundleProcessor.bundleAsJSONObject(bundle)
+        val notification = OSNotification(payload)
+
+        val notificationJob = OSNotificationGenerationJob(context)
```

Comment:
given that this file is on kotlin, can you use the apply block for building the object? apply block is the code style that we are following for "builder pattern"